docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,314 @@
+== Set the proxy for retrieving agent policies from {fleet}
+
+If there is a proxy between {agent} and {fleet}, specify proxy settings on the
+command line when you install {agent} and enroll in {fleet}. The settings you
+specify at the command line are added to the `fleet.yml` file installed on the
+system where the {agent} is running.
+
+NOTE: If {kib} is behind a proxy server, you'll still need to
+<<epr-proxy-setting,configure {kib} settings>> to access the package registry.
+
+The `enroll` and `install` commands accept the following flags:
+
+|===
+| CLI flag | Description
+
+|`--proxy-url <url>`
+|URL of the proxy server. The value may be either a complete URL or a
+`host[:port]`, in which case the http scheme is assumed.  The URL accepts optional
+username and password settings for authenticating with the proxy. For example:
+`http://<username>:<password>@<proxy host>/`.
+
+|`--proxy-disabled`
+|If specified, all proxy settings, including the `HTTP_PROXY` and `HTTPS_PROXY`
+environment variables, are ignored.
+
+|`--proxy-header <header name>=<value>`
+|Additional header to send to the proxy during CONNECT requests. Use the
+`--proxy-header` flag multiple times to add additional headers. You can use
+this setting to pass keys/tokens required for authenticating with the proxy.
+
+|===
+
+For example:
+
+[source,sh]
+----
+elastic-agent install -f --url="https://10.0.1.6:8220" --enrollment-token=TOKEN --proxy-url="http://10.0.1.7:3128" --fleet-server-es-ca="/usr/local/share/ca-certificates/es-ca.crt" --certificate-authorities="/usr/local/share/ca-certificates/fleet-ca.crt"
+----
+
+NOTE: This command requires default policies to be loaded in {fleet}. Default
+policies are loaded automatically when you visit {fleet} for the first time. If
+you're not sure whether default policies are loaded, log in to {kib} and go to
+*Management > {fleet}*.
+
+The command in the previous example adds the following settings to the
+`fleet.yml` policy on the host where {agent} is installed:
+
+[source,yaml]
+----
+fleet:
+  enabled: true
+  access_api_key: API-KEY
+  hosts:
+  - https://10.0.1.6:8220
+  ssl:
+    verification_mode: ""
+    certificate_authorities:
+    - /usr/local/share/ca-certificates/es-ca.crt
+    renegotiation: never
+  timeout: 10m0s
+  proxy_url: http://10.0.1.7:3128
+  reporting:
+    threshold: 10000
+    check_frequency_sec: 30
+  agent:
+    id: ""
+----
+
+[discrete]
+[[epr-proxy-setting]]
+== Set the proxy URL of the Elastic Package Registry
+
+{fleet} might be unable to access the Elastic Package Registry because {kib} is
+behind a proxy server.
+
+Also your organization might have network traffic restrictions that prevent {kib}
+from reaching the public Elastic Package Registry endpoints, like
+https://epr.elastic.co/[epr.elastic.co], to download package metadata and
+content. You can route traffic to the public endpoint of EPR through a network
+gateway, then configure proxy settings in the {kib} configuration file,
```

Comment:
Link to https://www.elastic.co/guide/en/kibana/current/fleet-settings-kb.html

Reply:
good idea